pytorch_blade/tests/mlir/test_disc_nn_ops.py

```diff
@@ -64,6 +65,7 @@ def log_softmax_func(x):
             log_softmax_func, x=torch.randint(-3, 3, [2, 3, 10, 4], dtype=torch.int32)
         )
 
+    @unittest.skipIf(not cuda_available, "Please fix incorrect results")
```

Comment:
Could you please provide more info about this failure？ Is it totally wrong or just less precision? Under the hood, the compiler itself uses polynomial approximation for the `mhlo.exp` , which may lead to some precision loss.

Reply:
```text
======================================================================
FAIL: test_softmax_func (tests.mlir.test_disc_nn_ops.TestDiscNNOps)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tianyou.gty/develop/opensource/BladeDISC/pytorch_blade/tests/mlir/test_disc_nn_ops.py", line 79, in test_softmax_func
    self._test_nn_ops(softmax)
  File "/home/tianyou.gty/develop/opensource/BladeDISC/pytorch_blade/tests/mlir/test_disc_nn_ops.py", line 23, in _test_nn_ops
    self._test_cvt_to_disc(nn_ops_func, test_data)
  File "/home/tianyou.gty/develop/opensource/BladeDISC/pytorch_blade/tests/mlir/testing_utils.py", line 54, in _test_cvt_to_disc
    self.assertEqual(output, result, rtol=rtol, atol=atol)
  File "/home/tianyou.gty/develop/opensource/BladeDISC/pytorch_blade/torch_blade/testing/common_utils.py", line 96, in assertEqual
    return super().assertEqual(x, y, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1165, in assertEqual
    super().assertTrue(result, msg=self._get_assert_msg(msg, debug_msg=debug_msg))
AssertionError: False is not true : Tensors failed to compare as equal!With rtol=1e-06 and atol=0.001, found 301056 element(s) (out of 301056) whose difference(s) exceeded the margin of error (including 301056 nan comparisons). The greatest difference was nan (0.004464285913854837 vs. nan), which occurred at index (0, 0, 0, 0).
```